homeassistant/components/fritz/sensor.py

```diff
@@ -295,26 +308,31 @@ class FritzBoxSensor(FritzBoxBaseEntity, SensorEntity):
     """Define FRITZ!Box connectivity class."""
 
     def __init__(
-        self, fritzbox_tools: FritzBoxTools, device_friendly_name: str, sensor_type: str
+        self,
+        fritzbox_tools: FritzBoxTools,
+        device_friendly_name: str,
+        sensor_type: FritzSensorEntityDescription,
     ) -> None:
         """Init FRITZ!Box connectivity class."""
-        self._sensor_data: SensorData = SENSOR_DATA[sensor_type]
+        self.entity_description: FritzSensorEntityDescription = sensor_type
```

Comment:
That will only set the type in `__init__`, not for any other method. To do that annotate it in the class scope. I.e.
```py
class FritzBoxSensor(FritzBoxBaseEntity, SensorEntity):
    """Define FRITZ!Box connectivity class."""

    entity_description: FritzSensorEntityDescription

    def __init__(self, ...): ...
```

```suggestion
        self.entity_description = description
```